mmv1/third_party/terraform/services/container/resource_container_cluster_test.go.erb

```diff
@@ -1539,6 +1539,108 @@ func TestAccContainerCluster_withNodeConfig(t *testing.T) {
 	})
 }
 
+func TestAccContainerCluster_withInsecureKubeletReadonlyPortEnabledInNodeConfig(t *testing.T) {
+	t.Parallel()
+	clusterName := fmt.Sprintf("tf-test-cluster-%s", acctest.RandString(t, 10))
+	networkName := acctest.BootstrapSharedTestNetwork(t, "gke-cluster")
+	subnetworkName := acctest.BootstrapSubnet(t, "gke-cluster", networkName)
+
+	acctest.VcrTest(t, resource.TestCase{
+		PreCheck:     func() { acctest.AccTestPreCheck(t) },
+		ProtoV5ProviderFactories: acctest.ProtoV5ProviderFactories(t),
+		CheckDestroy: testAccCheckContainerClusterDestroyProducer(t),
+		Steps: []resource.TestStep{
+			{
+				Config: testAccContainerCluster_withInsecureKubeletReadonlyPortEnabledInNodeConfig(clusterName, networkName, subnetworkName, "TRUE"),
```

Comment:
What would you think about adding steps to update this as well? It could be good to just make sure the bases are covered.

Reply:
I've added at least a basic one going from one setting to the other (though not with it missing the setting entirely) - let me know how you think it looks.

Reply:
Ack, should have realized, but this is the thing that is already known to be broken / not yet working (#19225), causing the failure [here](https://github.com/GoogleCloudPlatform/magic-modules/pull/11272#issuecomment-2312981818).

I'd tested locally, but only before adding in the second check.

```
    vcr_utils.go:167: Step 1/4 error: Check failed: Check 1/1 error: google_container_cluster.with_insecure_kubelet_readonly_port_enabled_in_node_config: Attribute 'node_pool_defaults.0.node_config_defaults.0.insecure_kubelet_readonly_port_enabled' expected "TRUE", got "FALSE"
```
This did help me realize that the way I switched the name for the _other_ test was wrong (it's testing "node pool defaults" not the "default pool", which is what _this_ one does). - I'm going to go back to something closer to the original name for _that_ (the other) test (`TestAccContainerCluster_withInsecureKubeletReadonlyPortEnabledDefaultsUpdates`), and also add some short comments before each test.

I could either
* switch this back to how it was
* mark this one as skipped and add a simpler variant that's separate
* leave this test as it is, but temporarily remove the second "Check" (that it's false), which I think will make it pass, but leave us with an evergreen test 😞 

For now, I'm going to do #3 assuming it passes locally; let me know if that seems reasonable and if the comments I've added make sense. I'm squashing all of these down, but let me know if it's easier if I push separate commits going forward for easier review if there are additional changes.

The fact that #3 passes without the `Check` makes me think maybe there's a missing idempotency check that should be in some of these tests too?

Time-permitting, I'm happy to try to take on 19225 and / or the autopilot one after this ships if you've got time to help provide some guidance, though in the end, may be easier / faster for someone from Google to handle those.

Reply:
And yeah, I'll test if it still bombs out on step 3/4 with the plancheck commented out -- if so, we might have to set the update test to skipped, and add separate create tests for true and / or false variants?

Reply:
Okay, I see what you mean. I don't think we need tests for the separate true / false variants, but we do need this field to either support in-place update or be marked ForceNew (if it can't be updated in place - we shouldn't intentionally ship a bug.) That's tricky because it would mean every iteration of the field would become ForceNew (including the ones that _can_ be updated in-place.)

Reply:
I think that at a minimum extracting the kubelet_config behavior from nodePoolUpdate & replicating it in the cluster Update method seems reasonable as a short-term solution here. There is precedent (in a way) with [changing node_version](https://github.com/GoogleCloudPlatform/magic-modules/blob/24542a8b0ad579c06980ab467ec694e58dff2ab5/mmv1/third_party/terraform/services/container/resource_container_cluster.go.erb#L3776-L3799) (though that uses cluster update instead of node pool update). (I'd also be fine with the full nodePoolUpdate change, though that would probably be best in a separate PR.)

Reply:
> but we do need this field to either support in-place update or be marked ForceNew (if it can't be updated in place - we shouldn't intentionally ship a bug.)

In practical terms, this is a bug that already exists in the code as it is now (https://github.com/hashicorp/terraform-provider-google/issues/19225), as the other fields in `kubelet_config` also cannot be successfully updated in place, which I think is why @trodge had previously approved shipping it with update broken for now. In other words, this update path doesn't work, but it doesn't introduce new broken behavior (other than introducing a new parameter for it). Updating it in place is probably not a common problem, but if someone does experience it, they'll just have permadrift, and could at least adjust the setting at the node-pool level via CLI or in the UI to reconcile the drift.

I had thought about doing `ForceNew`, but I'd argue that it is, actually, worse to ship an update that does that than shipping it as-is -- if someone accidentally does this while trying to disable the Kubelet read only port, they will potentially recreate their default nodepool, which is a much more time-consuming and potentially destructive operation. I think we'd have to force it on _any_ of the settings within the `kubelet_config` section.

> There is precedent (in a way) with [changing node_version](https://github.com/GoogleCloudPlatform/magic-modules/blob/24542a8b0ad579c06980ab467ec694e58dff2ab5/mmv1/third_party/terraform/services/container/resource_container_cluster.go.erb#L3776-L3799) (though that uses cluster update instead of node pool update). 

I'll take a look. That looks reasonable, however, I think there seems to be quite a lot going on in [nodePoolUpdate](https://github.com/GoogleCloudPlatform/magic-modules/blob/24542a8b0ad579c06980ab467ec694e58dff2ab5/mmv1/third_party/terraform/services/container/resource_container_node_pool.go.erb#L1349) including a whole section for `kubelet_config` that does exactly what we want (in fact, it _is_ exactly the same thing, since this is for a node_pool resource, and we're doing it for the `default-pool` nodepool resource created implicitly within the cluster resource):
https://github.com/GoogleCloudPlatform/magic-modules/blob/24542a8b0ad579c06980ab467ec694e58dff2ab5/mmv1/third_party/terraform/services/container/resource_container_node_pool.go.erb#L1778-L1802 -- pulling that plus the bit that handles waiting for the update etc., seems like quite a bit of code to extract.

So _if_ it's reasonable to call that directly, _and_ we can infer the reflected value in the (deprecated, in this case) fields directly within nodeConfig, it might not be too hard to "fix" this.

But I agree it would be better to fix that as a separate PR if at all possible, especially since it would very likely require fixing (and adding tests) for some of the other attributes as well, as well as maybe some other adjustments to the expand / flatten functions for one or more of them.

Reply:
Ok, so _if_ I were going to try to do that, would it be within here, right?:
https://github.com/GoogleCloudPlatform/magic-modules/blob/24542a8b0ad579c06980ab467ec694e58dff2ab5/mmv1/third_party/terraform/services/container/resource_container_cluster.go.erb#L3806-L3837

Would I just look for any changes within `node_config.0.kubelet_config` and run the update operation on that? Or take the narrow approach, and look for `node_config.0.kubelet_config.0.insecure_kubelet_readonly_port_enabled` and then call `updateNodePool` with at least some of the code around then waiting for that operation?

Reply:
Ok, I think I've got something working which I've just pushed up.

It may not be perfect, but seems to work in my functional tests, as well as in the unit test, and it's not a _crazy_ amount of extra code. Happy to make any minor adjustments for consistent style, avoiding hard-coded values (if there's a better way to get the name of the default pool without hardcoding it, let me know, ditto for how I'm setting the timeout, etc.), but I would rather if a broader fix for kubelet_config updates gets handled separately later.

The `cpu_manager_policy` permadrift still shows up in my functional testing; that's also related to a different bug, but setting it to `""` in the test should resolve that issue for now.

Reply:
I'm also pushing a slight variation that I can revert if you prefer the other way -- I think this will keep certain things at the top level scope of `kubelet_config` that will _hopefully_ make it easier to fix some of the other stuff in this section later?

Reply:
@melinath is that more of less what you were looking for, and do you have a sense which direction you prefer from the last 2 commits?

Reply:
I like the way you have it - seems like it could make the refactor more clear later. Thanks!